File: python/linked-list/implement.py
# Implementing a List Node
class Node:
    def __init__(self, val) -> None:
        self.val = val
        self.next = None

# Implementing a Linked List
class LinkedList:
    def __init__(self,head) -> None:
        self.head = head
    # count nodes of the list
    def size(self):
        current = self.head
        count = 0
        while current:
            count += 1
            current = current.next
        return count
    # return the last node of the list
    def getLast(self):
        last = self.head
        while last and last.next:
            last = last.next
        return last.val
    # print all the values of the nodes
    def traversal(self):
        current = self.head
        while current:
            print(current.val)
            current = current.next
    


# create nodes
node1 =  Node(1)
node2 =  Node(2)
node3 = Node(3)
# set pointers
node1.next = node2
node2.next = node3
# create linked list
linkedList = LinkedList(node1)
# acces the last node's value in the list 
print(linkedList.head.next.next.val) # expected: 3
# the size of the list
print(linkedList.size()) # expected output: 3
# last node val
print(linkedList.getLast()) # expected output: 3
# print all the values of the nodes
print(linkedList.traversal())

File: javascript/linked-list/implement.js
// Implementing a List Node 
class ListNode {
    constructor(val) {
        this.val = val
        this.next = null
    }
}

// Implementing a Linked List
class LinkedList {
    constructor(head = null) {
        this.head = head
    }
    // count nodes of the list
    size() {
        let count = 0
        let current = this.head
        while (current) {
            count++
            current = current.next
        }
        return count
    }
    // return the last node of the list
    getLast() {
        let last = this.head
        while (last && last.next) {
            last = last.next
        }
        return last
    }
    // print all the values of the nodes
    traversal() {
        let current = this.head
        while (current) {
            console.log(`${current.val}`)
            current = current.next
        }
        return;
    }
}
// create nodes
let node1 = new ListNode(1)
let node2 = new ListNode(2)
let node3 = new ListNode(3)
// set pointers
node1.next = node2
node2.next = node3
// create linked list
let list = new LinkedList(node1)
// acces the last node's value in the list 
console.log(list.head.next.next.val) // expected: 3
// return the size
console.log(list.size()) // expected: 3
// return the last node
console.log(list.getLast())

list.traversal()